sw/device/tests/entropy_src_fw_observe_many_contiguous.c

```diff
@@ -103,9 +108,63 @@ static status_t let_observe_fifo_overflow(uint32_t timeout_usec) {
   return OK_STATUS();
 }
 
+void ottf_external_isr(uint32_t *exc_info) {
+  dif_rv_plic_irq_id_t plic_irq_id;
+  CHECK_DIF_OK(dif_rv_plic_irq_claim(&rv_plic, kTopEarlgreyPlicTargetIbex0,
+                                     &plic_irq_id));
+
+  top_earlgrey_plic_peripheral_t peripheral =
+      top_earlgrey_plic_interrupt_for_peripheral[plic_irq_id];
+  CHECK(peripheral == kTopEarlgreyPlicPeripheralEntropySrc);
+
+  CHECK(plic_irq_id == kTopEarlgreyPlicIrqIdEntropySrcEsObserveFifoReady);
+
+  CHECK(!entropy_src_fifo_has_overflowed(),
+        "observe FIFO overflowed during collection");
+
+  uint32_t buf[32];
+  size_t len = ARRAYSIZE(buf);
+  CHECK_DIF_OK(
+      dif_entropy_src_observe_fifo_nonblocking_read(&entropy_src, buf, &len));
+
+  if (len > 0) {
+    CHECK_DIF_OK(
+        dif_entropy_src_observe_fifo_write(&entropy_src, buf, len, &len));
```

Comment:
what happens if the function does not manage to write the whole buffer? In this case, the test will not have pushed contiguous words to the FIFO. I think the code should at least check that possibility and explicitely handle it (error out?).

Reply:
Yeah okay, I've added an error to this test but I'm not sure how a real implementation would deal with that. What do you do if the noise source is still generating bits but there's no room in the FIFO for them? You've got to drop the bits somewhere...